clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2903,8 +2895,10 @@ def TryCallOp : CIR_CallOp<"try_call"> {
     [{
       $_state.addOperands(ValueRange{exception});
       $_state.addOperands(operands);
-      $_state.addAttribute("callee", callee);
-      $_state.addTypes(resType);
+      if (callee)
+        $_state.addAttribute("callee", callee);
+      if (resType && !resType.isa<VoidType>())
```

Comment:
The error you are getting is because cast/isa/dyn_cast now changed, see example on how to fix it here:
https://github.com/llvm/clangir/pull/692#discussion_r1656654311